tools/ibc-test/src/tests/ibc/sudt_erc20_transfer.rs

```diff
@@ -0,0 +1,577 @@
+        // ICS20TransferERC20 is bound to port transfer in axon chain
+        // initialization.
+        let port_b: PortId = "transfer".parse().unwrap();
+        let bootstrap_options = BootstrapChannelOptions::default()
+            .order(Order::Unordered)
+            .version("ics20".parse().unwrap())
+            .bootstrap_with_random_ids(false);
+        let channels = bootstrap_channel_with_connection(
+            &chains.handle_a,
+            &chains.handle_b,
+            connection,
+            &DualTagged::new(port_a).as_ref(),
+            &DualTagged::new(port_b).as_ref(),
+            bootstrap_options,
+        )
+        .unwrap();
+
+        // Send sudt ckb -> axon
+        log::info!("send ckb SUDT -> axon ERC20");
+        let metadata_script = packed::Script::new_builder()
+            .code_hash(ckb_config.client_code_hash.pack())
+            .hash_type(ScriptHashType::Type.into())
+            .args(CLIENT_TYPE_ARGS.as_bytes().pack())
+            .build();
+        let sdk_config = forcerelay_ckb_sdk::config::Config {
+            user_lock_script: AddressOrScript::Script(ibc_sudt_transfer_lock.clone().into()),
+            axon_metadata_type_script: AddressOrScript::Script(metadata_script.clone().into()),
+            channel_contract_type_id_args: ckb_config.channel_type_args.clone(),
+            packet_contract_type_id_args: ckb_config.packet_type_args.clone(),
+            channel_id: channels
+                .channel_id_a
+                .to_string()
+                .split('-')
+                .last()
+                .unwrap()
+                .parse()
+                .unwrap(),
+            confirmations: 1,
+        };
+        let (sudt_cell_dep, sudt_input, sudt_output, _amount) =
+            search_sudt_cells(&rt, &ckb_url).unwrap();
+        let sudt_type_script = sudt_output.type_().to_opt().unwrap();
+
+        // Create st-cell.
+        log::info!("create st-cell");
+        let st_cell_output = packed::CellOutput::new_builder()
+            .lock(ibc_sudt_transfer_lock.clone())
+            .type_(Some(sudt_type_script.clone()).pack())
+            .build_exact_capacity(Capacity::bytes(16).unwrap())
+            .unwrap();
+        let tx = TransactionBuilder::default()
+            .output(st_cell_output.clone())
+            .output_data(0u128.to_le_bytes().pack())
+            .cell_dep(sudt_cell_dep.clone())
+            .build();
+        let tx = complete_tx(
+            &ckb_url,
+            &tx,
+            ckb_sender_lock_script.clone(),
+            ckb_sender_key,
+        )
+        .unwrap();
+        let tx_hash = send_transaction(&ckb_url, tx).unwrap();
+        log::info!("deployed st-cell");
+        log::info!("send transfer packet");
+        let st_cell = packed::OutPoint::new_builder()
+            .tx_hash(tx_hash.pack())
+            .index(0u32.pack())
+            .build();
+        let denom = hex::encode(sudt_type_script.calc_script_hash().unpack());
```

Comment:
In the document [Understand IBC Denoms](https://tutorials.cosmos.network/tutorials/6-ibc-dev/), 
IBC denom is derived from the hash of the base_denom prepended with the path information, using the SHA256 hashing function.

#### Example
```js
// hash() representing a SHA256 hashing function returning a string
ibc_denom := 'ibc/' + hash('path' + 'base_denom')
          := 'ibc/' + hash(`{portID}/{channelID}/base_denom`)
```

I think the hash of `sudt_type_script` is only the 'base_denom' part, am I right?

Reply:
Tokens that traveled through different paths have different security guarantees.

Reply:
I guess so, `suit_type_script` is set as `base_denom`

Reply:
> #### Example
> ```js
> // hash() representing a SHA256 hashing function returning a string
> ibc_denom := 'ibc/' + hash('path' + 'base_denom')
>           := 'ibc/' + hash(`{portID}/{channelID}/base_denom`)
> ```

@blckngm said:
When using sudt-transfer-module as the sink zone, the `IBC denom hash` mode is preferred.